tests/test_insert_deletes_consolidate.cpp

```diff
@@ -159,209 +211,114 @@ void build_incremental_index(
-
   size_t dim, aligned_dim;
   size_t num_points;
 
   diskann::get_bin_metadata(data_path, num_points, dim);
   aligned_dim = ROUND_UP(dim, 8);
+  std::cout << "num_points = " << num_points << " "
+            << "dim = " << dim << std::endl;
+  std::cout << "aligned_dim = " << aligned_dim << std::endl;
 
   if (points_to_skip > num_points) {
     throw diskann::ANNException("Asked to skip more points than in data file",
                                 -1, __FUNCSIG__, __FILE__, __LINE__);
   }
-
-  if (max_points_to_insert == 0) {
-    max_points_to_insert = num_points;
+  if (max_points == 0) {
+    max_points = num_points;
   }
-
-  if (points_to_skip + max_points_to_insert > num_points) {
-    max_points_to_insert = num_points - points_to_skip;
-    std::cerr << "WARNING: Reducing max_points_to_insert to "
-              << max_points_to_insert
+  if (points_to_skip + max_points > num_points) {
+    max_points = num_points - points_to_skip;
+    std::cerr << "WARNING: Reducing max_points to " << max_points
               << " points since the data file has only that many" << std::endl;
   }
 
   using TagT = uint32_t;
   unsigned   num_frozen = 1;
   const bool enable_tags = true;
   const bool support_eager_delete = false;
-
-  auto num_frozen_str = getenv("TTS_NUM_FROZEN");
-
+  auto       num_frozen_str = getenv("TTS_NUM_FROZEN");
   if (num_frozen_str != nullptr) {
     num_frozen = std::atoi(num_frozen_str);
     std::cout << "Overriding num_frozen to" << num_frozen << std::endl;
   }
 
-  diskann::Index<T, TagT> index(diskann::L2, dim, max_points_to_insert, true,
-                                params, params, enable_tags,
-                                support_eager_delete, concurrent);
+  diskann::Index<T, TagT> index(diskann::L2, dim, max_points, true, params,
+                                params, enable_tags, support_eager_delete,
+                                concurrent);
 
-  size_t       current_point_offset = points_to_skip;
-  const size_t last_point_threshold = points_to_skip + max_points_to_insert;
+  const size_t last_point_threshold = points_to_skip + max_points;
 
-  if (beginning_index_size > max_points_to_insert) {
-    beginning_index_size = max_points_to_insert;
+  if (max_points * insert_percentage > max_points) {
+    insert_percentage = 1;
     std::cerr << "WARNING: Reducing beginning index size to "
-              << beginning_index_size
+              << max_points * insert_percentage
               << " points since the data file has only that many" << std::endl;
   }
-  if (checkpoints_per_snapshot > 0 &&
-      beginning_index_size > points_per_checkpoint) {
-    beginning_index_size = points_per_checkpoint;
-    std::cerr << "WARNING: Reducing beginning index size to "
-              << beginning_index_size << std::endl;
+  int64_t build_point = max_points * build_percentage;
+  if (build_point == max_points) {
+    build_point = max_points * build_percentage - 1;
```

Comment:
what's the error to use the full dataset to build?

Reply:
delete needs the empty slots, the size is coorespond with the size of max_points and  active points. When deleting points, make sure empty slots > delete_points, so build points should be less than max_points